WordPress/Classes/ViewRelated/Blog/KeyringAccountHelper.swift

```diff
@@ -0,0 +1,99 @@
+    var keyringConnection: KeyringConnection
+}
+
+@objc class KeyringAccountHelper: NSObject {
+
+    @objc class ValidationError: NSObject {
+        @objc let header: String
+        @objc let body: String
+        @objc let continueTitle: String
+        @objc let cancelTitle: String
+
+        @objc let continueURL: URL?
+
+        init(header: String, body: String, continueTitle: String, cancelTitle: String, continueURL: URL? = nil) {
+            self.header = header
+            self.body = body
+            self.continueTitle = continueTitle
+            self.cancelTitle = cancelTitle
+            self.continueURL = continueURL
+        }
+    }
+
+    /// Validate which account type has an error and return the information for an alert
+    ///
+    /// - Parameters:
+    ///     - connections: An array of `KeyringConnection` instances to normalize.
+    ///     - publicizeService: The publicize service for the fetched keyring connections.
+    ///
+    /// - Returns: An instance of `ValidationError` object, this is a plain object just with the information for an alert.
+    ///
+    @objc func validateConnections(_  connections: [KeyringConnection], with publicizeService: PublicizeService) -> ValidationError? {
+        let accounts = accountsFromKeyringConnections(connections, with: publicizeService)
+
+        if publicizeService.serviceID == PublicizeService.facebookServiceID, accounts.isEmpty {
+            return facebookPublicizeAlertFields()
+        }
+
+        return nil
+    }
+
+    /// Normalizes available accounts for a KeyringConnection and its `additionalExternalUsers`
+    ///
+    /// - Parameter connections: An array of `KeyringConnection` instances to normalize.
+    ///
+    /// - Returns: An array of `KeyringAccount` objects.
+    ///
+    func accountsFromKeyringConnections(_ connections: [KeyringConnection], with publicizeService: PublicizeService) -> [KeyringAccount] {
+        var accounts = [KeyringAccount]()
+
+        for connection in connections {
+            let acct = KeyringAccount(name: connection.externalDisplay, externalID: nil, externalIDForConnection: connection.externalID, keyringConnection: connection)
+
+            // Do not include the service if it only supports external users.
+            if !publicizeService.externalUsersOnly {
+                accounts.append(acct)
+            }
+
+            for externalUser in connection.additionalExternalUsers {
+                let acct = KeyringAccount(name: externalUser.externalName, externalID: externalUser.externalID, externalIDForConnection: externalUser.externalID, keyringConnection: connection)
+                accounts.append(acct)
+            }
+        }
+
+        return accounts
+    }
+
+}
+
+// MARK: - Alert Message Types
+
+private extension KeyringAccountHelper {
+
+    func facebookPublicizeAlertFields() -> ValidationError {
+        let alertHeaderMessage = NSLocalizedString("Not connected",
```

Comment:
Sorry I didn't notice this before, the “connected“ should be capitalized based on the screenshot in https://github.com/wordpress-mobile/WordPress-Android/issues/10278#issuecomment-522613358:

<img src="https://user-images.githubusercontent.com/1396552/63275712-035c0a00-c270-11e9-9156-cd03c616527b.png">